pilot/pkg/xds/filters/filters.go

```diff
@@ -185,51 +205,139 @@ var (
 )
 
 func buildHTTPMxFilter() *hcm.HttpFilter {
-	var vmConfig *v3.PluginConfig_VmConfig
+	httpMxConfigProto := &httpwasm.Wasm{
+		Config: &wasm.PluginConfig{
+			Vm:            constructVMConfig("/etc/istio/extensions/metadata-exchange-filter.compiled.wasm", "envoy.wasm.metadata_exchange"),
+			Configuration: util.MessageToAny(&protobuf.StringValue{Value: "{}"}),
+		},
+	}
+
+	return &hcm.HttpFilter{
+		Name:       MxFilterName,
+		ConfigType: &hcm.HttpFilter_TypedConfig{TypedConfig: util.MessageToAny(httpMxConfigProto)},
+	}
+}
+
+func BuildTCPStatsFilter(class networking.ListenerClass) *listener.Filter {
+	js, err := json.Marshal(statsConfigForClass(class))
+	if err != nil {
+		log.Errorf("failed to marshal stats config: %v", err)
+		return nil
+	}
+
+	vmConfig := constructVMConfig("/etc/istio/extensions/stats-filter.compiled.wasm", "envoy.wasm.stats")
+	root := rootIDForClass(class)
+	vmConfig.VmConfig.VmId = "tcp_" + root
+
+	wasmConfig := &wasmfilter.Wasm{
+		Config: &wasm.PluginConfig{
+			RootId:        root,
+			Vm:            vmConfig,
+			Configuration: util.MessageToAny(&protobuf.StringValue{Value: string(js)}),
+		},
+	}
+
+	return &listener.Filter{
+		Name:       StatsFilterName,
+		ConfigType: &listener.Filter_TypedConfig{TypedConfig: util.MessageToAny(wasmConfig)},
+	}
+}
+
+func BuildHTTPStatsFilter(class networking.ListenerClass) *hcm.HttpFilter {
+	js, err := json.Marshal(statsConfigForClass(class))
+	if err != nil {
+		log.Errorf("failed to marshal stats config: %v", err)
+		return nil
+	}
 
+	vmConfig := constructVMConfig("/etc/istio/extensions/stats-filter.compiled.wasm", "envoy.wasm.stats")
+	root := rootIDForClass(class)
+	vmConfig.VmConfig.VmId = root
+
+	wasmConfig := &httpwasm.Wasm{
+		Config: &wasm.PluginConfig{
+			RootId:        root,
+			Vm:            vmConfig,
+			Configuration: util.MessageToAny(&protobuf.StringValue{Value: string(js)}),
+		},
+	}
+
+	return &hcm.HttpFilter{
+		Name:       StatsFilterName,
+		ConfigType: &hcm.HttpFilter_TypedConfig{TypedConfig: util.MessageToAny(wasmConfig)},
+	}
+}
+
+func rootIDForClass(class networking.ListenerClass) string {
+	switch class {
+	case networking.ListenerClassSidecarInbound:
+		return "stats_inbound"
+	default:
+		return "stats_outbound"
+	}
+}
+
+func statsConfigForClass(class networking.ListenerClass) statsConfig {
+	switch class {
+	case networking.ListenerClassSidecarInbound:
+		return statsConfig{
+			Metrics: []*metricsConfig{{
+				Dimensions: map[string]string{
+					"destination_cluster": "node.metadata['CLUSTER_ID']",
```

Comment:
Add a bug to make them first class.

Reply:
I thikn they are already first classed https://github.com/istio/proxy/pull/3229. It was a mistake that they were not removed from inbound in this PR: https://github.com/istio/istio/pull/31226/files